app/logic/courseProposals.py

```diff
@@ -19,3 +24,13 @@ def getProposalData(user):
         "term":course.term,
         "status":course.status.status}
     return courseDict
+
+def deleteProposal(courseID):
+    """Deletes proposal of ID passed in. Removes foreign keys first.
+    Key Dependencies: QuestionNote, CourseQuestion, CourseParticipant,
+    CourseInstructor, Note"""
+    course = Course.get(Course.id == courseID)
+    questions = CourseQuestion.select().where(CourseQuestion.course == course)
+    notes = [note for note in Note.select(Note.id).where(QuestionNote.question.in_(questions)).distinct().join(QuestionNote)]
```

Comment:
Rather than this list comprehension, you can just directly convert it with `list(your query)`. I'm surprised that just using the query doesn't work in the delete on line 36 though.